src/components/SiteDetail.js

```diff
@@ -43,14 +41,13 @@ class SiteDetail extends Component {
     this.loadSiteData();
   }
 
-  async loadSiteData() {
+  loadSiteData() {
     const { selectSite } = this.props;
     const { loadedSite } = this.state;
 
     if (selectSite) {
       if (!loadedSite || (loadedSite && loadedSite.id !== selectSite.id)) {
-        const { data: loadedSite } = await summary.get(`/sites/${selectSite.id}/`);
-        this.setState({ loadedSite });
+        this.setState({ loadedSite: selectSite });
```

Comment:
A thought:  I'm not sure if you've used React Hooks before, but this would be one component where I would encourage you to take a swing at it:

- you have a single element in state
- every change to a prop causes something to re-render


it's as simple as starting with 

```js
import React, {useState} from 'react'

(...)
const Componet = ({selectSite}) => {
  const [loadedSite, setLoadedSite] = useState(null)

  if (selectSite && (!loadedSite || (loadedSite.id !== selectSite.id))) {
    setLoadedSite(selectsite)
  } 

(...)```